docs/source/delta-spark-connect.md

```diff
@@ -0,0 +1,119 @@
+## Motivation
+
+Delta Connect is expected to bring the same benefits as Spark Connect:
+
+1. Easier upgrading to more recent versions of Spark and <Delta>, as the client interface is completely decoupled from the server.
+2. Simpler integration of Spark and <Delta> with developer tooling. IDEs no longer have to integrate with the full Spark and <Delta> implementation, and instead can integrate with a thin-client.
+3. Support for languages other than Java/Scala and Python. Clients "merely" have to generate Protocol Buffers and therefore become simpler to implement.
+4. Spark and <Delta> will become more stable, as user code is no longer running in the same JVM as Spark's driver.
+5. Remote connectivity. Code can run anywhere now, as there is a gRPC layer between the user interface and the driver.
+
+## How to start the Spark Server with Delta
+
+1. Download `spark-4.0.0-preview1-bin-hadoop3.tgz` from [Spark 4.0.0 First Preview](https://archive.apache.org/dist/spark/spark-4.0.0-preview1).
+
+2. Start the Spark Connect server with the <Delta> Connect plugins:
+
+```bash
+sbin/start-connect-server.sh \ 
+  --packages org.apache.spark:spark-connect_2.13:4.0.0-preview1,io.delta:delta-connect-server_2.13:4.0.0-preview1,io.delta:delta-connect-common_2.13:4.0.0-preview1,com.google.protobuf:protobuf-java:3.25.1 \ 
+  --conf "spark.sql.extensions=io.delta.sql.DeltaSparkSessionExtension" \
+  --conf "spark.sql.catalog.spark_catalog=org.apache.spark.sql.delta.catalog.DeltaCatalog" \
+  --conf "spark.connect.extensions.relation.classes=org.apache.spark.sql.connect.delta.DeltaRelationPlugin" \
+  --conf "spark.connect.extensions.command.classes=org.apache.spark.sql.connect.delta.DeltaCommandPlugin"
+```
+
+## How to use the Python Spark Connect Client with Delta
+
+The <Delta> Connect Python client is included in the same PyPi package as <Delta> Spark.
+
+1. `pip install pyspark==4.0.0`
+
+2. `pip install delta-spark==4.0.0`
+
+3. There is no difference in usage compared to the classic way. We just need to pass in a remote `SparkSession` (instead of a local one) to the `DeltaTable` API.
+
+An example:
+
+```python
+import io.delta.tables._
+import org.apache.spark.sql.functions._
+
+val deltaTable = DeltaTable.forName(spark, "my_table")
+
+deltaTable.update(
+  condition = expr("id % 2 == 0"),
+  set = Map("id" -> expr("id + 100")))
+```
+
+## How to use the Scala Spark Connect Client with Delta
+
+.. note:: These steps are going to be simplified with final Spark 4.0 and Delta 4.0
+
+Make sure you are using Java 17!
+
+1. `export SPARK_REMOTE="sc://localhost:15002"`
```

Comment:
Where does the port number here come from? I remember in some tests using port 15003? I do not see it being passed in the server initialization above.

By the way, we need to test all steps in the documentation.

Reply:
@andreaschat-db It's from https://spark.apache.org/docs/4.0.0-preview1/spark-connect-overview.html#configure-client-server-connection